californium-core/src/test/java/org/eclipse/californium/core/test/lockstep/ObserveClientSideTest.java

```diff
@@ -333,44 +356,52 @@ public void testBlockwiseObserveAndNotificationWithoutBlockwise() throws Excepti
 
 		// Send observe request
 		client.sendRequest(request);
-
 		// Expect observe request
 		server.expectRequest(CON, GET, path).storeMID("OBS_REQ").storeToken("OBS_TOK").go();
-		// Send complete response and blockwise response at the same time \o/ !!
-		server.sendEmpty(ACK).loadMID("OBS_REQ").go();
-		int mid_block = 111;
-		server.sendResponse(CON, CONTENT).loadToken("OBS_TOK").observe(1).mid(mid_block).block2(0, true, 16)
-				.payload(respPayload.substring(0, 16)).go();
-		int mid_notif = 222;
-		server.sendResponse(CON, CONTENT).loadToken("OBS_TOK").observe(2).mid(mid_notif)
-				.payload(notifpayload)
-				.go();
-
-		// Expect for block request and ACKs
-		server.expectEmpty(ACK, mid_block).go();
-		server.expectRequest(CON, GET, path).storeMID("BLOCK_TOK").storeToken("SECOND_BLOCK").block2(1, false, 16)
-				.go();
-		server.expectEmpty(ACK, mid_notif).go();
-
-		// Send next block
-		server.sendResponse(ACK, CONTENT).loadMID("BLOCK_TOK").loadToken("SECOND_BLOCK").block2(1, true, 16)
-				.payload(respPayload.substring(16, 32)).go();
 
+		// send blockwise response
+		server.sendResponse(ACK, CONTENT).loadMID("OBS_REQ").loadToken("OBS_TOK").block2(0, true, 16).observe(1)
+				.payload(respPayload.substring(0, 16)).go();
+		// Expect next block request
+		server.expectRequest(CON, GET, path).storeMID("SECOND_BLOCK_MID").storeToken("SECOND_BLOCK_TOK")
+				.block2(1, false, 16).go();
+
+		// During block transfer send a complete (none blockwise) response
+		int new_mid = 222;
+		server.sendResponse(CON, CONTENT).loadToken("OBS_TOK").observe(2).mid(new_mid).payload(notifpayload).go();
+		server.expectEmpty(ACK, new_mid++).go();
 		// Check that we get the most recent response (with the higher observe
 		// option value)
 		Response response = request.waitForResponse();
 		assertResponseContainsExpectedPayload(response, notifpayload);
 
+		// Send next block
+		server.sendResponse(CON, CONTENT).loadMID("SECOND_BLOCK_MID").loadToken("SECOND_BLOCK_TOK").block2(1, true, 16)
+				.payload(respPayload.substring(16, 32)).go();
+		// ensure client don't ask for block anymore
+		server.expectEmpty(RST, (Integer) server.get("SECOND_BLOCK_MID")).go();
+		// TODO ensure that blockdata buffer is cleared in blockwiselayer...
```

Comment:
I didn't a find a good way to check blockdata in blockwiselayer is cleaned.
I previously workaround that by sending a GET request and see if it's work but now this test is in #357.